tests/runtime/btf

```diff
@@ -97,3 +97,9 @@ PROG kprobe:kvm:x86_emulate_insn { $ctxt = (struct x86_emulate_ctxt *) arg0; pri
 EXPECT Attaching 1 probe...
 TIMEOUT 2
 REQUIRES lsmod | grep '^kvm'
+
+NAME btf_type_tag_access
+PROG BEGIN { printf("SUCCESS %d\n", curtask->parent->pid); exit() }
+EXPECT_REGEX SUCCESS [0-9][0-9]*
+TIMEOUT 5
+REQUIRES_FEATURE btf
```

Comment:
Note this requires a clang built linux kernel. So this is probably a noop for the reasonably close future. Perhaps leave a comment?

Reply:
Oh? Are most kernels (I guess I also in our CI) not built with clang?

Reply:
Most (if not all) kernels in wild are built with gcc. Only folks using clang built Linux I know of is meta and Google. Support came from google.

In kernel source the btf type tag annotation is ifdefd out for gcc. Look for the __has_attribute compiler built in. So it looks like a regular field in btf when kernel is compiled with gcc.